dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2723,14 +2723,19 @@ void MergeTreeData::movePartitionToDisk(const ASTPtr & partition, const String &
     if (!disk)
         throw Exception("Disk " + name + " does not exists on policy " + storage_policy->getName(), ErrorCodes::UNKNOWN_DISK);
 
-    for (const auto & part : parts)
+    parts.erase(std::remove_if(parts.begin(), parts.end(), [&](auto part_ptr) {
+        return part_ptr->disk->getName() == disk->getName();
+    }), parts.end());
+
+    if (!parts.empty())
     {
-        if (part->disk->getName() == disk->getName())
-            throw Exception("Part " + part->name + " already on disk " + name, ErrorCodes::UNKNOWN_DISK);
+        if (!movePartsToSpace(parts, std::static_pointer_cast<const DiskSpace::Space>(disk)))
+            throw Exception("Cannot move parts because moves are manually disabled.", ErrorCodes::ABORTED);
+    }
+    else
+    {
+        LOG_DEBUG(log, "No parts of partition " << partition_id << " to move to disk " << disk->getName());
```

Comment:
We should throw an exception here, all parts already on the destionation disk.

Reply:
3debdc211904eb0b5fe6cca1a4af99d8db635892